go/rtl/auth.go

```diff
@@ -119,7 +120,10 @@ func (s *AuthSession) Do(result interface{}, method, ver, path string, reqPars m
 			return fmt.Errorf("response error. status=%s. error parsing error body", res.Status)
 		}
 
-		return fmt.Errorf("response error. status=%s. error=%s", res.Status, string(b))
+		return ResponseError{
```

Comment:
Can we pass in `res.Status` too as a separate field? Then in the `Error()` method we can reuse it so the error message doesn't technically change (incase anyone was parsing it)